samples/bluetooth/hci_rpmsg/src/main.c

```diff
@@ -242,13 +257,75 @@ static void hci_rpmsg_send(struct net_buf *buf)
 #if defined(CONFIG_BT_CTLR_ASSERT_HANDLER)
 void bt_ctlr_assert_handle(char *file, uint32_t line)
 {
-	BT_ASSERT_MSG(false, "Controller assert in: %s at %d", file, line);
+#if defined(CONFIG_BT_HCI_VS_FATAL_ERROR)
+	/* Disable interrupts, this is unrecoverable */
+	(void)irq_lock();
+
+	/* Generate an error event only when IPC service endpoint is already bound. */
+	if (ipc_ept_ready) {
+		/* Prepare vendor specific HCI debug event */
+		struct net_buf *buf;
+
+		buf = hci_vs_err_assert(file, line);
+		if (buf == NULL) {
+			/* Send the event over rpmsg */
+			hci_rpmsg_send(buf, HCI_FATAL_ERR_MSG);
+		} else {
+			LOG_ERR("Can't create Fatal Error HCI event: %s at %d", __FILE__, __LINE__);
+		}
+	} else {
+		LOG_ERR("IPC endpoint is not redy yet: %s at %d", __FILE__, __LINE__);
+	}
+
+	LOG_ERR("Halting system");
+
+	while (true) {
+	};
+#else
+	LOG_ERR("Controller assert in: %s at %d", file, line);
+#endif /* CONFIG_BT_HCI_VS_FATAL_ERROR */
 }
 #endif /* CONFIG_BT_CTLR_ASSERT_HANDLER */
 
+#if defined(CONFIG_BT_HCI_VS_FATAL_ERROR)
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *esf)
+{
+	LOG_PANIC();
+
+	/* Disable interrupts, this is unrecoverable */
+	(void)irq_lock();
+
+	/* Generate an error event only when there is a stack frame and IPC service endpoint is
+	 * already bound.
+	 */
+	if (esf != NULL && ipc_ept_ready) {
+		/* Prepare vendor specific HCI debug event */
+		struct net_buf *buf;
+
+		buf = hci_vs_err_stack_frame(reason, esf);
+		if (buf != NULL) {
+			hci_rpmsg_send(buf, HCI_FATAL_ERR_MSG);
+		} else {
+			LOG_ERR("Can't create Fatal Error HCI event.\n");
+		}
+	}
+
+	LOG_ERR("Halting system");
+
+	while (true) {
+	};
+
+	CODE_UNREACHABLE;
+}
+#endif /* CONFIG_BT_HCI_VS_FATAL_ERROR */
+
 static void hci_ept_bound(void *priv)
 {
 	k_sem_give(&ipc_bound_sem);
+#if defined(CONFIG_BT_CTLR_ASSERT_HANDLER) ||                                  \
+	defined(CONFIG_BT_HCI_VS_FATAL_ERROR)
```

Comment:
On line 41 you have these on a single line:
```c
#if defined(CONFIG_BT_CTLR_ASSERT_HANDLER) || defined(CONFIG_BT_HCI_VS_FATAL_ERROR)
```

Reply:
Aligned with line 41.